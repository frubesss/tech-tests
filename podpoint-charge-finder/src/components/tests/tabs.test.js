import React from 'react';
import Tabs from '../tabs'
import renderer from 'react-test-renderer'
import {shallow} from "enzyme/build/index";

describe('Tabs Component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Tabs buttonClick={() => {}}/>);
    });

    it('Snapshot Test', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with the favourites tab active', () => {
        expect(wrapper.find('.btn-group').childAt(1).hasClass('active')).toBe(true);
    });

    describe('Tabs', () => {

        it('should render with the favourites tab active', () => {
            expect(wrapper.find('.btn-group').childAt(1).hasClass('active')).toBe(true);
        });

        describe('click', () => {

            beforeEach(() => {
                wrapper = shallow(<Tabs buttonClick={() => {}} buttonActive={'NEARBY'}/>);
                wrapper.find('.btn-group').childAt(0).simulate('click');
            });

            it('should change to the selected tab when a new tab is selected', () => {
                expect(wrapper.find('.btn-group').childAt(0).hasClass('active')).toBe(true);
            });

            it('should remove active class from the previous tab when a new tab is selected', () => {
                expect(wrapper.find('.btn-group').childAt(1).hasClass('active')).toBe(false);
            });

        });

    })


});
