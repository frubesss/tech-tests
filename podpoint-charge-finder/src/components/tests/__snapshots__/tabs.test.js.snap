// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component Snapshot Test 1`] = `
<div>
  <div
    className="btn-group"
  >
    <button
      className="btn pill"
      onClick={[Function]}
    >
      NEARBY
    </button>
    <button
      className="btn pill active"
      onClick={[Function]}
    >
      FAVOURITES
    </button>
  </div>
</div>
`;
